lib/errors.js

```diff
@@ -33,6 +33,7 @@ createError('FST_ERR_DEC_MISSING_DEPENDENCY', `The decorator is missing dependen
 */
 createError('FST_ERR_HOOK_INVALID_TYPE', `The hook name must be a string`, 500, TypeError)
 createError('FST_ERR_HOOK_INVALID_HANDLER', `The hook callback must be a function`, 500, TypeError)
+createError('FST_ERR_HOOK_INVALID_FUNCTION', `You are using an async function with three arguments, 'next' is not needed.`, 500, TypeError)
```

Comment:
```suggestion
createError('FST_ERR_HOOK_INVALID_FUNCTION', `Async function has too many arguments. Async hooks should not use the `next` argument.`, 500, TypeError)
```

Suggesting this phrasing because:
- It's nicer to avoid using the word "you" in error messages
- Hooks can have 3 or 4 arguments